hazelcast-jet-core/src/main/java/com/hazelcast/jet/Observable.java

```diff
@@ -0,0 +1,222 @@
+ * gets a reference to its equivalent {@code Observable} (through
+ * {@link JetInstance#getObservable(String) JetInstance.getObservable()})
+ * and registers the first {@link Observer} on it (through
+ * {@link Observable#addObserver(Observer) Observable.addObserver()})
+ * or when the job containing the sink for it starts executing.
+ * <p>
+ * The {@code Ringbuffer} must be explicitly destroyed when it's no longer
+ * in use, or data will be retained in the cluster. This is done via the
+ * {@link #destroy() Observable.destroy()} method. Note: even if the
+ * {@code Observable} POJO gets lost and its underlying {@code Ringbuffer}
+ * floats around in the cluster, it's still possible to manually destroy
+ * it later by creating another {@code Observable} instance with the same
+ * name and calling {@code destroy()} on that.
+ * <p>
+ * <strong>Important:</strong> The same {@code Observable} must
+ * <strong>not</strong> be used again in a new job since this will cause
+ * completion events interleaving and causing data loss or other unexpected
+ * behaviour. Same goes for using one observable name in multiple
+ * {@link Sinks#observable(String) observable sinks} in the same job.
+ * <p>
+ * @param <T> type of the values in the sequence
+ *
+ * @since 4.0
+ */
+public interface Observable<T> extends Iterable<T> {
+
+    /**
+     * Returns the name identifying this particular observable.
+     *
+     * @return name of observable
+     */
+    @Nonnull
+    String name();
+
+    /**
+     * Registers an {@link Observer} to this {@code Observable}. It will
+     * receive all events currently in the backing {@link Ringbuffer} and
+     * then continue receiving any future events.
+     *
+     * @return registration ID associated with the added {@code Observer},
+     * can be used to remove the {@code Observer} later
+     */
+    @Nonnull
+    UUID addObserver(@Nonnull Observer<T> observer);
+
+    /**
+     * Removes a previously added {@link Observer} identified by its
+     * assigned registration ID. A removed {@code Observer} will not get
+     * notified about further events.
+     */
+    void removeObserver(@Nonnull UUID registrationId);
+
+    /**
+     * Set the capacity of the {@code Observable}. Being backed by a
+     * {@link Ringbuffer}, an {@code Observable}'s capacity is basically
+     * the {@code Ringbuffer}'s capacity, which defaults to 10,000.
+     * <p>
+     * This method can be called only before the {@code Ringbuffer} gets
+     * created. This means before any {@link Observer}s are added to the
+     * {@code Observable} and before any jobs containing
+     * {@link com.hazelcast.jet.pipeline.Sinks#observable(String) observable
+     * sinks} (with the same observable name) are submitted for execution.
+     * <p>
+     * If the {@code Ringbuffer} has already been created, then this
+     * method will throw an {@link IllegalStateException}.
+     */
+    void setCapacity(int capacity);
```

Comment:
can we add a note about the default capacity?

Reply:
There is, in the first paragraph of the method's Javadoc.